sql/src/main/java/io/crate/analyze/PrivilegesAnalyzer.java

```diff
@@ -95,17 +101,27 @@ private static void ensureUserManagementEnabled(@Nullable User user) {
         return privilegeTypes;
     }
 
-    private void validateTableNames(List<QualifiedName> tableNames) {
-        tableNames.forEach(t -> schemas.getTableInfo(TableIdent.fromIndexName(t.toString())));
+    private void validateTableNames(List<String> tableNames) {
+        tableNames.forEach(t -> {
+            TableIdent ident = TableIdent.fromIndexName(t);
+            List<String> schemaInfo = Collections.singletonList(ident.schema());
+            validateSchemaNames(schemaInfo);
+            schemas.getTableInfo(ident);
+        });
     }
 
-    private void validateSchemaNames(List<QualifiedName> schemaNames) {
-        schemaNames.forEach(s ->
-            schemas.validateSchemaName(s.toString())
+    private void validateSchemaNames(List<String> schemaNames) {
+        String message = "GRANT/DENY/REVOKE Privileges on information_schema is not supported";
+        schemaNames.forEach(s -> {
+                if (InformationSchemaInfo.NAME.equals(s)) {
+                    throw new UnsupportedFeatureException(message);
+                }
+                schemas.validateSchemaName(s);
+            }
         );
     }
 
-    private void validatePrivilegeIdents(Privilege.Clazz clazz, List<QualifiedName> idents) {
+    private void validatePrivilegeIdents(Privilege.Clazz clazz, List<String> idents) {
```

Comment:
pls rename and refactor this method to `List<String> convertAndValidateIdents(Privilege.Clazz clazz, List<QualifiedName> idents)`, do the conversion inside this method, validate and return the strings. 
change the `privilegeTypesToPrivileges` method to accept list of strings as idents, and remove the current conversion there.